rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -522,6 +551,232 @@ void save_callsFederationSupportSave() throws IOException {
+
+        @BeforeEach
+        void setUp() {
+            long rskExecutionBlockNumber = 1000L;
+            long rskExecutionBlockTimestamp = 10L;
+            BlockHeader blockHeader = new BlockHeaderBuilder(mock(ActivationConfig.class))
+                .setNumber(rskExecutionBlockNumber)
+                .setTimestamp(rskExecutionBlockTimestamp)
+                .build();
+            rskExecutionBlock = Block.createBlockFromHeader(blockHeader, true);
+
+            activeFederation = FederationTestUtils.getErpFederation(federationConstantsMainnet.getBtcParams());
+            List<UTXO> activeFederationUTXOs = BitcoinTestUtils.createUTXOs(10, activeFederation.getAddress());
+            proposedFederation = new P2shErpFederationBuilder().build();
+
+            federationSupport = mock(FederationSupport.class);
+            when(federationSupport.getActiveFederation()).thenReturn(activeFederation);
+            when(federationSupport.getActiveFederationAddress()).thenReturn(activeFederation.getAddress());
+            when(federationSupport.getActiveFederationBtcUTXOs()).thenReturn(activeFederationUTXOs);
+            when(federationSupport.getProposedFederation()).thenReturn(Optional.of(proposedFederation));
+
+            Coin feePerKb = Coin.valueOf(1000);
+            when(feePerKbSupport.getFeePerKb()).thenReturn(feePerKb);
+
+            rskTxHash = PegTestUtils.createHash3(1);
+            rskTx = mock(Transaction.class);
+            when(rskTx.getHash()).thenReturn(rskTxHash);
+
+            BridgeEventLogger bridgeEventLogger = new BridgeEventLoggerImpl(bridgeMainNetConstants, activationsAfterForks, logs, signatureCache);
+
+            bridgeSupport = bridgeSupportBuilder
+                .withBridgeConstants(bridgeMainNetConstants)
+                .withEventLogger(bridgeEventLogger)
+                .withProvider(bridgeStorageProvider)
+                .withActivations(activationsAfterForks)
+                .withFederationSupport(federationSupport)
+                .withFeePerKbSupport(feePerKbSupport)
+                .withExecutionBlock(rskExecutionBlock)
+                .build();
+        }
+
+        @Test
+        void processSvpFundTransactionWithoutSignatures_whenProposedFederationDoesNotExist_throwsIllegalStateException() {
+            // arrange
+            when(federationSupport.getProposedFederation()).thenReturn(Optional.empty());
+
+            // act & assert
+            IllegalStateException exception = assertThrows(IllegalStateException.class,
+                () -> bridgeSupport.processSvpFundTransactionUnsigned(rskTx));
+
+            String expectedMessage = "Proposed federation should be present when processing SVP fund transaction.";
+            assertEquals(expectedMessage, exception.getMessage());
+        }
+
+        @Test
+        void processSvpFundTransactionWithoutSignatures_createsExpectedTransactionAndSavesTheHashInStorageEntryAndPerformsPegoutActions() throws Exception {
```

Comment:
No other unhappy path? Not enough UTXOs in the Bridge to create the fund tx?

Reply:
not enough utxos should throw an `InsufficientMoneyException`, but from `completeTx` method.
I mean, that test should be part of Wallet testing class.

The only exception being thrown from this method implementation is when not having a proposed fed

Reply:
Disagree, I think that if we are testing behavior it's a good test case. Regardless of where the error happens

Reply:
Done